src/BinSkim.Rules/PERules/BA6001.EnableSourceLink.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+
+using Microsoft.CodeAnalysis.BinaryParsers;
+using Microsoft.CodeAnalysis.BinaryParsers.ProgramDatabase;
+using Microsoft.CodeAnalysis.IL.Sdk;
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.CodeAnalysis.Sarif.Driver;
+
+namespace Microsoft.CodeAnalysis.IL.Rules
+{
+    [Export(typeof(Skimmer<BinaryAnalyzerContext>)), Export(typeof(ReportingDescriptor)), Export(typeof(IOptionsProvider))]
+    public class EnableSourceLink : WindowsBinaryAndPdbSkimmerBase, IOptionsProvider
+    {
+        /// <summary>
+        /// BA6001
+        /// </summary>
+        public override string Id => RuleIds.EnableSourceLink;
+
+        /// <summary>
+        /// Enable SourceLink.
+        /// </summary>
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString { Text = RuleResources.BA6001_EnableSourceLink_Description };
+
+        protected override IEnumerable<string> MessageResourceNames => new string[]
+        {
+            nameof(RuleResources.BA6001_Pass),
+            nameof(RuleResources.BA6001_Error)
+        };
+
+        private const string AnalyzerName = RuleIds.EnableSourceLink + "." + nameof(EnableSourceLink);
+
+        public override void Initialize(BinaryAnalyzerContext context)
+        {
+            base.Initialize(context);
+        }
+
+        public override AnalysisApplicability CanAnalyzePE(PEBinary target, PropertiesDictionary policy, out string reasonForNotAnalyzing)
+        {
+            reasonForNotAnalyzing = null;
+            return AnalysisApplicability.ApplicableToSpecifiedTarget;
+        }
+
+        public override void AnalyzePortableExecutableAndPdb(BinaryAnalyzerContext context)
+        {
+            if (HasSourceLink(context))
+            {
+                context.Logger.Log(this,
+                    RuleUtilities.BuildResult(ResultKind.Pass, context, null,
+                    nameof(RuleResources.BA6001_Pass),
+                    context.TargetUri.GetFileName()));
+            }
+            else
+            {
+                context.Logger.Log(this,
+                    RuleUtilities.BuildResult(ResultKind.Fail, context, null,
```

Comment:
>BuildResult [](http://example.com/codeflow?start=34&length=11)

So for the failure case, you don't pass ResultKind.Fail, you call a different overload that accepts the failure severity. For this check, we need a default severity of 'warning', as 'error' in our domain is a serious 'must fix' designation.

Another option for this rule is to make it disabled by default, and require people to opt in. I tend to think we should try to make the 'warning' error level work.

RuleUtilities.BuildResult(FailureLevel.Warning